runtime/commonMain/src/kotlinx/serialization/internal/Enums.kt

```diff
@@ -1,22 +1,16 @@
 /*
- * Copyright 2017-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ * Copyright 2017-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
  */
 
 package kotlinx.serialization.internal
 
 import kotlinx.serialization.*
-import kotlin.jvm.*
-import kotlin.reflect.*
 
 @InternalSerializationApi
-public class EnumDescriptor @JvmOverloads constructor(
+public class EnumDescriptor constructor(
     name: String,
-    values: Array<String> = emptyArray()
-) : SerialClassDescImpl(name) {
-
-    init {
-        values.forEach { addElement(it) }
-    }
+    elementsCount: Int = 1
```

Comment:
do we really need default value here?

Reply:
It's not very convenient when it's used in tests like this: https://github.com/kotlin/kotlinx.serialization/blob/d0fb63d5002c85632cfb524aabbb0a8883242c32/runtime/commonTest/src/kotlinx/serialization/features/BinaryPayloadExampleTest.kt#L15 (and user code)

Once this constructor won't be user-facing, there will be no need in default value